### PR TITLE
Add support for properties in parent classes

### LINKED ISSUE #1579: [autogrid] show properties from parent classes
### Describe your motivation

When using AutoGrid in the Pet Clinic applications, veterinarians don't have first and last name since they come from the `Person` parent class. They should be available in the grid by default.

See https://github.com/cromoteca/hilla-petclinic/blob/10fc9164cbb29629b35bebe8e2cdf52ab86bdd4d/src/main/java/org/springframework/samples/petclinic/vet/Vet.java#L46

Hilla 2.3.1

### PR DESCRIPTION
**Pull request by cromoteca**
Fixes #1579

### CODE REVIEW

File: packages/ts/react-crud/src/property-info.ts [hunk 1]
```diff
@@ -59,34 +60,47 @@ export function _generateHeader(path: string): string {
...
-      type,
-      meta,
-    };
-  });
+  return (
+    propertyIds
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      .map((name) => [name, modelInstance[name]])
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      .filter(([, m]) => m?.[_meta])
```
Comment (sissbruecker) [inline packages/ts/react-crud/src/property-info.ts:83]: Is the filter here necessary? Metadata is in general optional and there are specific conditions that determine whether it is generated at all.

File: packages/ts/react-crud/src/property-info.ts [hunk 2]
```diff
@@ -59,34 +60,47 @@ export function _generateHeader(path: string): string {
     .replace(/^./u, (match) => match.toUpperCase());
 }
 
-export const getPropertyIds = (model: DetachedModelConstructor<AbstractModel>): string[] =>
-  Object.keys(Object.getOwnPropertyDescriptors(model.prototype)).filter((p) => p !== 'constructor');
+export const getPropertyIds = (model: DetachedModelConstructor<AbstractModel>): string[] => {
+  const propertyIds: string[] = [];
+
+  for (let proto = model; proto !== ObjectModel; proto = Object.getPrototypeOf(proto)) {
+    // the choice of putting parent properties at the beginning is deliberate, `push` could be used instead
```
Comment (sissbruecker) [inline packages/ts/react-crud/src/property-info.ts:67]: Explaining what the alternatives could be is unnecessary and just adds cognitive load. Better to just explain what it does.

File: packages/ts/react-crud/src/property-info.ts (hunk 1 above)
Comment (cromoteca) [reply packages/ts/react-crud/src/property-info.ts:83]: I'm filtering out those who don't have metadata at all

File: packages/ts/react-crud/src/property-info.ts (hunk 2 above)
Comment (cromoteca) [reply packages/ts/react-crud/src/property-info.ts:67]:
```suggestion
    // parent properties are added at the beginning
```

Comment (sissbruecker) [reply packages/ts/react-crud/src/property-info.ts:67]: Looks good 👍

File: packages/ts/react-crud/src/property-info.ts (hunk 1 above)
Comment (cromoteca) [reply packages/ts/react-crud/src/property-info.ts:83]: The original code (before this PR) assumes that `meta` is there. It maybe is, but TypeScript complains if I remove this check.

Comment (sissbruecker) [reply packages/ts/react-crud/src/property-info.ts:83]: I just tried to remove it, I can not observe any errors.

**Review by sissbruecker — APPROVED** [review]

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**